Nodejs/Product/Npm/SPI/NpmGetCatalogCommand.cs

```diff
@@ -65,115 +65,182 @@ internal class NpmGetCatalogCommand : NpmCommand, IPackageCatalog {
-                                try {
-                                    builder.Name = (string)module["name"];
-                                    if (string.IsNullOrEmpty(builder.Name)) {
-                                        continue;
-                                    }
-
-                                    builder.AppendToDescription((string)module["description"] ?? string.Empty);
-
-                                    var time = module["time"];
-                                    if (time != null) {
-                                        builder.AppendToDate((string)time["modified"]);
-                                    }
-
-                                    var distTags = module["dist-tags"];
-                                    if (distTags != null) {
-                                        var latestVersion = distTags
-                                            .OfType<JProperty>()
-                                            .Where(v => (string)v.Name == "latest")
-                                            .Select(v => (string)v.Value)
-                                            .FirstOrDefault();
-
-                                        if (!string.IsNullOrEmpty(latestVersion)) {
-                                            try {
-                                                builder.LatestVersion = SemverVersion.Parse(latestVersion);
-                                            } catch (SemverVersionFormatException) {
-                                                OnOutputLogged(String.Format(Resources.InvalidPackageSemVersion, latestVersion, builder.Name));
-                                            }
-                                        }
-                                    }
-
-                                    var versions = module["versions"];
-                                    if (versions != null) {
-                                        builder.AvailableVersions = GetVersions(versions);
-                                    }
-
-                                    AddKeywords(builder, module["keywords"]);
-
-                                    AddAuthor(builder, module["author"]);
-
-                                    AddHomepage(builder, module["homepage"]);
-
-                                    var package = builder.Build();
-
-                                    InsertCatalogEntry(db, package);
-                                } catch (InvalidOperationException) {
-                                    // Occurs if a JValue appears where we expect JProperty
-                                } catch (ArgumentException) {
-                                    OnOutputLogged(string.Format(Resources.ParsingError, builder.Name));
-                                    if (!string.IsNullOrEmpty(builder.Name)) {
-                                        var package = builder.Build();
-                                        InsertCatalogEntry(db, package);
-                                    }
```

Comment:
We should continue catching the ArgumentException - occasionally npm changes things about their schema (or some odd package gets published that is contrary to the entire schema) that causes us to parse things incorrectly. For instance, at some point a single package was appearing in the catalog with an array of homepages rather than a single list. 

In these cases, we want to be able to fail gracefully with a small note about what failed rather than failing to retrieve the entire catalog.